pages/[id]/bridge.tsx

```diff
@@ -101,20 +132,66 @@ const Bridge = () => {
   };
 
   const handleBridge = () => {
-    return bridgeTarget === BridgeTarget.L2 ? bridgeToL2!() : bridgeToL1!();
+    if (bridgeTarget === BridgeTarget.L2) {
+      if (!bridgeToL2) throw new Error('bridgeToL2 is not defined');
+      bridgeToL2();
+    } else {
+      if (!bridgeToL1) throw new Error('bridgeToL1 is not defined');
+      bridgeToL1();
+    }
   };
 
-  const source =
-    bridgeTarget === BridgeTarget.L2
-      ? { ...l1Config, ...l1, fee: fees?.l1 || BigInt(0) }
-      : { ...l2Config, ...l2, fee: fees?.l2 || BigInt(0) };
-  const target =
+  const formatError = (e: Error | null): ErrorReturnType => {
+    if (e === null) return { errorType: undefined, errorReason: undefined };
+    const errorTypes = [
+      {
+        errorType: ErrorType.UserRejectedRequest,
+        errorSearchString: 'User rejected the request.',
+        prettyReason: `Error: User rejected the request.`,
+      },
+      {
+        errorType: ErrorType.ERC20AmountError,
+        errorSearchString: 'ERC20: transfer amount exceeds balance',
```

Comment:
Where is the message for the error defined? What if we change the message changes slightly? Will the search then fail for this error.

Reply:
> What if we change the message changes slightly? Will the search then fail for this error.

the message is a viem error. if viem changes their error strings, we'll have to change this error parser. with viem pinned I think we're safe.